tests/test_utils.py

```diff
@@ -298,11 +306,536 @@ def check_multistream_classic(fnames):
+                        annual_mean = {}
+                        for key, df in data.items():
+                            annual_mean[key] = pd.DataFrame(df[vic_var].
+                                                            groupby(df.index.
+                                                            hour).mean())
+
+                        df = pd.DataFrame({key: d[vic_var] for key, d in
+                                          annual_mean.items() if vic_var in d})
+
+                        for key, series in df.iteritems():
+                            series.plot(linewidth=compare_data_dict[key]
+                                        ['linewidth'],
+                                        ax=axarr[i],
+                                        color=compare_data_dict[key]['color'],
+                                        linestyle=compare_data_dict[key]
+                                        ['linestyle'],
+                                        zorder=compare_data_dict[key]
+                                        ['zorder'])
+
+                        axarr[i].legend(loc='upper left')
+                        axarr[i].set_ylabel('%s ($W/{m^2}$)' % variable_name,
+                                            size=fs)
+                        axarr[i].set_xlabel('Time of Day (Hour)', size=fs)
+                        axarr[i].set_xlim([0, 24])
+                        axarr[i].xaxis.set_ticks(np.arange(0, 24, 3))
+
+                    # save plot
+                    plotname = '%s_%s.png' % (lat, lng)
+                    print(lat)
+                    print(lng)
+                    os.makedirs(os.path.join(plot_dir, 'annual_mean'),
+                                exist_ok=True)
+                    savepath = os.path.join(plot_dir, 'annual_mean', plotname)
+                    plt.savefig(savepath, bbox_inches='tight', dpi=dpi)
+                    print("saved figure to %s" % savepath)
+
+                    plt.clf()
+                    plt.close()
+
+            if 'monthly_mean_diurnal_cycle' in plots_to_make:
+
+                # make monthly mean diurnal cycle plots
+                with plt.rc_context(dict(sns.axes_style("whitegrid"),
+                                    **sns.plotting_context("notebook"))):
+                    f, axarr = plt.subplots(4, 12, figsize=(35, 7),
+                                            sharex=True,
+                                            sharey=True)
+
+                    months = ['January', 'February', 'March', 'April', 'May',
+                              'June', 'July', 'August', 'September',
+                              'October', 'November', 'December']
+
+                    for i, (vic_var, variable_name) in enumerate(vars.items()):
+                        for j in range(12):
+
+                            # calculate monthly mean diurnal cycle
+                            monthly_mean = {}
+                            for (key, df) in data.items():
```

Comment:
I think the for loop over months (line 782) and `data.items()` (line 786) can be switched. The current implementation is doing the expensive groupby-apply-combine steps in the groupby 11 times more than it needs to.